src/GridFS/Bucket.php

```diff
@@ -0,0 +1,210 @@
+            }
+        }
+        if (isset($options['writeConcern'])) {
+            if (! $options['writeConcern'] instanceof WriteConcern) {
+                throw new InvalidArgumentTypeException('"writeConcern" option', $options['writeConcern'], 'MongoDB\Driver\WriteConcern');
+            } else {
+                $collectionOptions['writeConcern'] = $options['writeConcern'];
+            }
+        }
+        $this->databaseName = (string) $databaseName;
+        $this->options = $options;
+
+        $this->filesCollection = new Collection(
+            $manager,
+            sprintf('%s.%s.files', $this->databaseName, $options['bucketName']),
+            $collectionOptions
+        );
+        $this->chunksCollection = new Collection(
+            $manager,
+            sprintf('%s.%s.chunks', $this->databaseName, $options['bucketName']),
+            $collectionOptions
+        );
+    }
+    /**
+     * Opens a Stream for reading the contents of a file specified by ID.
+     *
+     * @param ObjectId $id
+     * @return Stream
+     */
+    public function openDownloadStream(ObjectId $id)
+    {
+        fopen('gridfs://$this->databaseName/$id', 'r');
+    }
+      /**
+       * Downloads the contents of the stored file specified by id and writes
+       * the contents to the destination Stream.
+       * @param ObjectId  $id           GridFS File Id
+       * @param Stream    $destination  Destination Stream
+       */
+    public function downloadToStream(ObjectId $id, $destination)
+    {
+        $result = $this->filesCollection->findOne(['_id' => $id]);
+        if ($result == null) {
+            return;
+        }
+        if ($result->length == 0){
+            return;
+        }
+
+        $n=0;
+        $results = $this->chunksCollection->find(['files_id' => $result->_id]);
+        foreach ($results as $chunk) {
+            if ($chunk->n != $n) {
+                return;
+            }
+            fwrite($destination, $chunk->data);
+            $n++;
+        }
+    }
+    /**
+     * Return the chunkSizeBytes option for this Bucket.
+     *
+     * @return integer
+     */
+    public function getChunkSizeBytes()
+    {
+        return $this->options['chunkSizeBytes'];
+    }
+    /**
+     * Opens a Stream for writing the contents of a file.
+     *
+     * @param string   $filename     file to upload
+     * @param array    $options      Stream Options
+     * @return Stream  uploadStream
+     */
+    public function openUploadStream($filename, array $options = [])
+    {
+        $this->ensureIndexes();
+        $options = [
+            'filesCollection' => $this->filesCollection,
+            'chunksCollection' => $this->chunksCollection,
+            'chunkSizeBytes' => $this->getChunkSizeBytes()
+            ];
+        $context = stream_context_create(['gridfs' => $options]);
+        return fopen('gridfs://$this->databaseName/$filename', 'w', false, $context);
```

Comment:
`$this->databaseName` and `$filename` aren't going to be resolved within this single-quoted string. If you want to use double-quoted strings, you can surround these with `{}` for clarity (see: [complex expressions](http://php.net/manual/en/language.types.string.php#language.types.string.parsing.complex)); otherwise, I'd opt for `sprintf()`.